Source/Particles/Pusher/GetAndSetPosition.H

```diff
@@ -0,0 +1,76 @@
+#ifndef WARPX_PARTICLES_PUSHER_GETANDSETPOSITION_H_
+#define WARPX_PARTICLES_PUSHER_GETANDSETPOSITION_H_
+
+#include <limits>
+#include <WarpXParticleContainer.H>
+#include <AMReX_REAL.H>
+
+#ifndef WARPX_RZ
+
+/* \brief Extract the particle's coordinates from the ParticleType struct `p`,
+ *        and stores them in the variables `x`, `y`, `z`. */
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void GetPosition(
+    amrex::Real& x, amrex::Real& y, amrex::Real& z,
+    const WarpXParticleContainer::ParticleType& p)
+{
+#if (AMREX_SPACEDIM==3)
+    x = p.pos(0);
+    y = p.pos(1);
+    z = p.pos(2);
+#else
+    x = p.pos(0);
+    y = std::numeric_limits<Real>::quiet_NaN();
+    z = p.pos(1);
+#endif
+}
+
+/* \brief Set the particle's coordinates in the ParticleType struct `p`,
+ *        from their values in the variables `x`, `y`, `z`. */
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void SetPosition(
+    WarpXParticleContainer::ParticleType& p,
+    const amrex::Real x, const amrex::Real y, const amrex::Real z)
```

Comment:
Are you passing `x`, `y` and `z` by value as `const` here so that developers see this won't be modified? Or is there another reason (e.g., it might help the compiler for optimization)?

Reply:
It can help avoid accidentally modifying `x`.   It's fine with or without `const`.  I have no strong preference either way.  Note that for C compatibility, in C++ `void f(const double)` has the same signature as `void f(double)`.